### ISSUE TITLE
Docking up while being target painted causes a segfault

### ISSUE THREAD

**Issue by jdhirst** [EvEmu-Project/evemu_Crucible#39]
From AlasiyaEvE created by [ryllae](https://github.com/ryllae): zhyrohaad/AlasiyaEvE#108  

Equip a fleet republic target painter, activate on a player character in high sec.     Have the other toon dock up.  SEGFAULT.

**Comment by jdhirst** [comment]
Rats could be shooting you while warping or session changing, so I assume you'd have the same problem there either way.  It's just easier for me to test docking.

**Comment by jdhirst** [comment]
target manager has been updated, and seems to be working correctly from my testing.
update pushed to main server.

**Comment by jdhirst** [comment]
Nope.  I undocked a Hawk and a Noctis.  Had the Hawk target paint the Noctis.  Crashes when the Noctis docks.

**Comment by jdhirst** [comment]
that tells me that module code thats supposed to be doing "ship status" updates (jump, dock, warp, etc) are NOT processing correctly.
i'll look into this tonite.

*edit*
it's ANY module activated on a docking ship, not just painters
*edit again*
and its not just docking.
my status update processor was faulty and incomplete.
did some more work to it.
test again and report when ya get time, pls.

**Comment by jdhirst** [comment]
must be something funny on main server, as i cannot reproduce this on dev

**Comment by jdhirst** [comment]
Still crashes.

**Comment by jdhirst** [comment]
I think your latest comment confirmed this, but it's still broken.

**Comment by jdhirst** [comment]
Working correctly on dev `eb777b`.

**Comment by jdhirst** [comment]
This has been updated and should work for all possible scenarios

***EDIT - UPDATE***
after digging into this a bit more, i have found TargetManager code is old, outdated, and about half of it's responses to client are being ignored, as it was written for the Apoc version.
am currently rewriting functions and methods, and updating responses.

**Comment by jdhirst** [comment]
touch

**Comment by jdhirst** [comment]
Not really.  TargMgr is separate system.
However, never expected rats targeting while docking, so never tested that scenario.

**Comment by jdhirst** [comment]
main server updated

**Comment by jdhirst** [comment]
Haven't tested target painting, but rats acquiring a lock while docking segfaults. https://github.com/zhyrohaad/AlasiyaEvE/issues/75  Related?

**Comment by jdhirst** [comment]
I hadn't thought of it, but you're right.
However it's not coded that way, in contrast to my generic module "target destroyed" call.   The dock, jump,warp, etc, is specific, but should not be.  This highlights a shortcoming in the code and need to be addressed.
This and the referenced module code are top priority.

**Comment by jdhirst** [comment]
Still crashes.

**Comment by zhyrohaad** [comment]
has this been looked at or tested since merge to official branch?
prefer testing on staging, as that code has more fixes (but possibly has more errors)
